modules/home/workstation/i3/polybar.lib.nix

```diff
@@ -0,0 +1,29 @@
+{ config, pkgs, lib, ... }:
+
+let
+  launchScript = ''
+    # Terminate already running bar instances
+    ${pkgs.procps}/bin/pkill polybar
+
+    # Wait until the processes have been shut down
+    while ${pkgs.procps}/bin/pgrep polybar >/dev/null; do ${pkgs.coreutils}/bin/sleep 1; done
+
+    if type "xrandr"; then
+      for m in $(${pkgs.xorg.xrandr}/bin/xrandr --query | ${pkgs.gnugrep}/bin/grep " connected" | ${pkgs.coreutils}/bin/cut -d" " -f1); do
+        MONITOR=$m polybar --reload top &
+      done
+    else
+      polybar --reload top &
+    fi
```

Comment:
Keep only the truthy part of the if condition here. `if type "xrandr"` is useless here, as you're calling xrandr directly from the Nix store.

Reply:
Do I really need the condition then?